discourse_rock/patches/discourse-charm.patch

```diff
@@ -0,0 +1,18 @@
+--- a/lib/tasks/discourse-charm.rake
++++ b/lib/tasks/discourse-charm.rake
+@@ -0,0 +1,15 @@
++# frozen_string_literal: true
++
++desc "Check if a user exists for given email address"
++task "users:exists", [:email] => [:environment] do |_, args|
++  email = args[:email]
++  if !email || email !~ /@/
++    puts "ERROR: Expecting rake users:exists[some@email.com]"
++    exit 1
++  end
++  if User.find_by_email(email)
++    puts "ERROR: User with email #{email} not found"
++    exit 1
```

Comment:
At the moment there is no way of differentiating user not found from a failure (except matching on the exact error string).

I believe this should be handled some other way, let's say using another status code for `user not found` of having status code 0 for `user not found` and returning some meaningful message in stdout (true/false or something like that).

Reply:
I still think the issue is there. If this task fails for whatever reason, the event create user will fail with:
`event.fail(f"User with email {email} does not exist")`

Reply:
i have addressed this now, by returning a different error code and managing this in the except